### PR TITLE
Added a port for xmlsec library

### PR DESCRIPTION
**Pull request by SvenPStarFinanz**
Should work on Windows, Linux, OSX.

### CODE REVIEW

**Comment by SvenPStarFinanz** [comment]
Have to work on static triplet.

File: ports/xmlsec/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,25 @@
+include(vcpkg_common_functions)
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO lsh123/xmlsec
+    REF xmlsec-1_2_28
+    SHA512 fb0c775f6455ce5a5579a69bb91d60fe90c023e538c32bdf2a70aa413a53b22ef938349a3ce6b42bb23f8f70b4e00f1b9917f877487bb1507c927ec70c3d95f5
+    HEAD_REF master
+)
+
```
Comment (Voskrese) [inline ports/xmlsec/portfile.cmake]: vcpkg_check_linkage(ONLY_STATIC_LIBRARY)

**Comment by Voskrese** [comment]
@SvenPStarFinanz  
why shut down? dynamically also compiled through the original configurator

need to watch variables

**Comment by Voskrese** [comment]
debug dynamic
```
set(STATIC 0)
set(WITH_DL 1)
```

```
# .\configure.txt
# This file is generated automatically by configure.js.

XMLSEC_VERSION_MAJOR=1
XMLSEC_VERSION_MINOR=2
XMLSEC_VERSION_SUBMINOR=28
BASEDIR=..
XMLSEC_SRCDIR=..\src
APPS_SRCDIR=..\apps
BINDIR=binaries
WITH_CRYPTO=mscrypto
WITH_DEFAULT_CRYPTO=mscrypto
WITH_OPENSSL=0
WITH_OPENSSL_VERSION=XMLSEC_OPENSSL_
WITH_NSS=0
WITH_MSCRYPTO=1
WITH_MSCNG=0
WITH_LIBXSLT=1
WITH_ICONV=1
WITH_NT4=1
UNICODE=1
DEBUG=1
WERROR=0
CC=cl.exe
CFLAGS=/MDd
STATIC=0
WITH_DL=1
PREFIX=E:\tools\vcpkg\installed\x64-windows\debug
BINPREFIX=$(PREFIX)\bin
INCPREFIX=$(PREFIX)\include
LIBPREFIX=$(PREFIX)\lib
SOPREFIX=$(PREFIX)\bin
INCLUDE=$(INCLUDE);.
LIB=$(LIB);.
CRUNTIME=/MDd
```

**Comment by SvenPStarFinanz** [comment]
I had some problems with git handling. Resubmitted in #7196 now.

**Pull request closed without merging**